File: server/model/Album.java

package server.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Document(collection = "albums")
public class Album {
    @Id
    private String id;
    private String name;
    private String description;
    private String coverPhotoId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
}

File: server/model/Album.js

const mongoose = require('mongoose');

const albumSchema = new mongoose.Schema({
  name: { type: String, required: true },
  description: String,
  coverPhotoId: { type: mongoose.Schema.Types.ObjectId, ref: 'Photo' },
}, { timestamps: true });

module.exports = mongoose.model('Album', albumSchema);
